emulator/src/kreator/assembler.rs

```diff
@@ -0,0 +1,101 @@
+use std::fmt;
+
+struct Assembler {
+    code: Vec<String>,
+    program_counter: u16,
+}
+
+impl Assembler {
+    fn new(input_code: &str) -> Assembler {
+        let mut lines: Vec<String> = Vec::new();
+        for line in input_code.split("\n") {
+            let line_parts = line.trim().split(";").collect::<Vec<&str>>();
+            if line_parts[0].len() != 0 {
+                lines.push(String::from(line_parts[0].trim()));
+            }
+        }
+        Assembler { 
+            code: lines,
+            program_counter: 0,
+        }
+    }
+}
+
+fn to_binary(instruction: &String) -> u8 {
```

Comment:
to_binary is an incorrect function name. This function does not convert the instruction to binary, it converts it to the corresponding machine code. Converting to binary would simply mean converting the string to its binary representation.